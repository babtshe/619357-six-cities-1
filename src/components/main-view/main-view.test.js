import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MainView from './main-view';

jest.mock(`leaflet`);


it(`MainView should render correctly`, ()=>{
  const mockCards = [
    {
      name: `test1`,
      type: `apartment`,
      price: 1,
      rating: 1,
      id: 1,
      image: `test`,
      premium: true,
      location: [52.3909553943508, 4.85309666406198],
    },
    {
      name: `test2`,
      type: `room`,
      price: 1,
      id: 2,
      rating: 1,
      image: `test`,
      bookmarked: true,
      location: [52.369553943508, 4.85309666406198],
    }
  ];
  const mockCity = {name: `test`, location: [0, 0]};
  const renderer = new ShallowRenderer();
  const tree = renderer
  .render(
      <MainView
        offers = {mockCards}
        setCity = {()=>{}}
        cities = {[mockCity]}
        currentCity = {mockCity}
      />);

  expect(tree).toMatchSnapshot();
});
