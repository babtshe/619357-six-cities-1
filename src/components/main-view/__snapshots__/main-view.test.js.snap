// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView should render correctly 1`] = `
<main
  className="page__main page__main--index"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="cities tabs"
  >
    <CitiesList
      cities={
        Array [
          Object {
            "location": Array [
              0,
              0,
            ],
            "name": "test",
          },
        ]
      }
      currentCity={
        Object {
          "location": Array [
            0,
            0,
          ],
          "name": "test",
        }
      }
      setCity={[Function]}
    />
  </div>
  <div
    className="cities__places-wrapper"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          2
           place
          s
           to stay in 
          test
        </b>
        <withRouter(WithQuery) />
        <withRouter(WithSorting)
          classPrefix="cities"
          offers={
            Array [
              Object {
                "id": 1,
                "image": "test",
                "location": Array [
                  52.3909553943508,
                  4.85309666406198,
                ],
                "name": "test1",
                "premium": true,
                "price": 1,
                "rating": 1,
                "type": "apartment",
              },
              Object {
                "bookmarked": true,
                "id": 2,
                "image": "test",
                "location": Array [
                  52.369553943508,
                  4.85309666406198,
                ],
                "name": "test2",
                "price": 1,
                "rating": 1,
                "type": "room",
              },
            ]
          }
        />
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map map"
        >
          <CityMap
            activeLocation={null}
            cityLocation={
              Array [
                0,
                0,
              ]
            }
            locations={
              Array [
                Array [
                  52.3909553943508,
                  4.85309666406198,
                ],
                Array [
                  52.369553943508,
                  4.85309666406198,
                ],
              ]
            }
          />
        </section>
      </div>
    </div>
  </div>
</main>
`;
